congame-doc/introduction.scrbl

```diff
@@ -1,37 +1,530 @@
+are other expressions that can be used to insert the various other form elements like checkboxes and
+dropdown lists.}
+          
+@item{The @racket[(set! first-name (input-text))] expression tells Conscript, “when this form is
+submitted, set the value in the @racket[_first-name] variable to the whatever the user has entered
+in @racket[(input-text)]. }
+
+@item{The @racket[submit-button] expression inserts the button the user can click to submit all the
+information they have entered into the form elements.}
+
+@item{All the interactive form elements are wrapped in a @racket[form] expression to keep them
+grouped together.}
+
+]
+
+Next, write the final step by adding these lines to the end of your file:
+
+@codeblock[#:keep-lang-line? #f]|{
+#lang conscript/local
+
+(defstep (thank-you)
+  @md{
+    # Good job, @first-name
+
+    Thank you for participating in our survey despite
+    being @number->string[age] years old.
+  })
+}|
+
+We’re doing something else interesting here: we’re inserting our variables directly into the text in
+the @racket[md] expression! We’ve told Conscript to record @racket[_first-name] and @racket[_age] in
+the previous step of the study, but in addition to storing those values for later analysis, we can
+make use of those values while the user is still participating. There are many ways we might want to
+do this; in this case, we’re simply displaying the values back to the user.
+
+@margin-note{Note that since we used @racket[input-number] to assign a value to @racket[_age], we
+must use @racket[number->string] to convert that numeric value to text if we want to display it
+within @racket[md].}
+
+Finally, we’ll define the study as a whole by tying all the steps together in a transition graph:
+
+@codeblock[#:keep-lang-line? #f]|{
+#lang conscript/local
+
+(defstudy simple-survey
+  [description --> age-name-survey --> thank-you]
+  [thank-you --> thank-you])
+
+(provide simple-survey)
+
+}|
+
+We’ve now told Conscript that our study is named @racket[simple-survey], and that it consists of
+three steps @racket[description], @racket[age-name-survey] and @racket[thank-you] in that order.
+Because every step must have a transition, even the last one, we add @racket[[thank-you -->
+thank-you]] to tell Conscript that that step simply transitions to itself.
+
+The last line is not technically required at @emph{this} point, but it will be needed later when we
+upload our study to a Congame server. Without it, the server will not be able to access the study
+bound to the @racket[simple-study] identifier. 
+
+@margin-note{It’s a good practice to add @racket[(provide _studyname)] following every
+@racket[defstudy] expression in your study program. Alternatively, you could simply add a single
+@racket[(provide (all-defined-out))] expression, which will cover all the studies defined in the
+file with a single line of code.}
+
+Combining all these snippets, the code should look like the below example. Go ahead and save it as
```

Comment:
Notice that you used different indentation for different `md`s, which is fine. It is important to have the same level of indentation within a given `md`, otherwise you get surprises. It is probably worth adding a note --- TODO comment --- to highlight this potential issue here (should they mess up) and add an example with a broken `md` and screenshot to the tutorial, where the first line is indented 4 spaces and a later paragraph 6 spaces, which will turn it into a quote and break a bunch of stuff. Showing this example will prime them so that when (not if) they hit this bug, they know where to go to find out what the problem is.

Reply:
I debated whether I should use `md` or `html` here. The indenting concern (which I’d forgotten about) makes me suspect `html` would ultimately be less confusing for the quick intro, and mesh better with the Overview (which has a short explainer on HTML). What do you think, should we switch to `html` here, or just add the caveats/examples around indentation in `md` — or both? 

In any case I will make a note to add warnings and examples of this caveat to the `md` and `md*` functions in the module reference.

Reply:
We'll stick with `md` despite this gotcha, students just find it so much easier than `html` and they come across that type of syntax in more places including Git. `html` is for later, when one needs to add classes and ids to elements.